Homework/HW01.hs

```diff
@@ -0,0 +1,64 @@
+module Homework.HW01 where
+
+-- Homework Url https://www.cis.upenn.edu/~cis194/spring15/hw/01-intro.pdf
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the last digit from a number
+lastDigit :: Integer -> Integer
+lastDigit num = num `mod` 10
+
+-- Drop the last digit from a number
+dropLastDigit :: Integer -> Integer
+dropLastDigit num = num `div` 10 
+
+-- Exercise 2 -----------------------------------------
+
+toRevDigits :: Integer -> [Integer]
+toRevDigits num 
+    | num > 0           = lastDigit num : (toRevDigits . dropLastDigit) num
+    | otherwise         = []
+
+-- Exercise 3 -----------------------------------------
+
+-- Double every second number in a list starting on the left.
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther [] = []
+doubleEveryOther num_list
+    | length num_list > 1  = head num_list: 2* head (tail num_list) : (doubleEveryOther . tail .tail) num_list
+    | otherwise            = head num_list: []
+
+-- Exercise 4 -----------------------------------------
+
+getSumOfDigits :: Integer -> Integer
+getSumOfDigits n
+    | n>10              = (n `mod` 10) + (n `div` 10)
```

Comment:
i intentionally left out the recursion bit, because in this particular case the double of any single digit number is never going to exceed **`18`** so this specific use case was only for 2 digit numbers that too below 18 but you are right I should have taken care of generally scenario as well, what if its 3 digit. Updated it.